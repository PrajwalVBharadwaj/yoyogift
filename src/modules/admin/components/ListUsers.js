import React, { useState, useEffect } from "react";
import axiosWrapper from "../../../apis/axiosCreate";
import { Table, Column } from "react-virtualized";
import "react-virtualized/styles.css";

export default props => {
  const [users, updateUsers] = useState(null);
  const [, update] = useState(null);
  window.addEventListener("resize", update);
  useEffect(() => {
    if (users === null)
      axiosWrapper.get("/users").then(response => {
        updateUsers(response.data);
      });
  });
  if (users === null) {
    return <p>Loading...</p>;
  }
  const width = window.innerWidth;
  const height = (window.innerHeight * 80) / 100;
  return (
    <Table
      //   style={{ height: "80vh" }}
      width={width}
      height={height}
      headerHeight={40}
      rowHeight={40}
      rowCount={users.length}
      rowGetter={({ index }) => users[index]}
    >
      <Column label="First Name" dataKey="first_name" width={width / 4} />
      <Column label="Last Name" dataKey="last_name" width={width / 4} />
      <Column label="Points" dataKey="balance_points" width={width / 4} />
      <Column label="E-mail ID" dataKey="email" width={width / 4} />
    </Table>
  );
};
