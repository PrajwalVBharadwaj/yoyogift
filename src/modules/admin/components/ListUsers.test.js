import React, { useState, useEffect } from "react";
import { shallow, userDetails } from "../../../Enzyme";
import axiosWrapper from "../../../apis/axiosCreate";
import ListUsers from "./ListUsers";

describe("Testing List users component", () => {
  it("Snapshot testign", () => {
    axiosWrapper.get = jest.fn().mockImplementationOnce(() =>
      Promise.resolve({
        data: {
          users: [
            {
              fisrt_name: "a",
              last_name: "b",
              email: "c",
              balance_points: 10
            }
          ]
        }
      })
    );
    const wrapper = shallow(<ListUsers />);
    expect(wrapper).toMatchSnapshot();
  });
  // it("When users are loading", () => {
  //   jest.spyOn(React, "useState").mockImplementationOnce(() => {
  //     useState({
  //       users: [
  //         {
  //           fisrt_name: "a",
  //           last_name: "b",
  //           email: "c",
  //           balance_points: 10
  //         }
  //       ]
  //     });
  //   });
  //   jest.spyOn(React, "useEffect").mockImplementationOnce(() => {
  //     useEffect(() => {
  //       return 1;
  //     });
  //   });
  //   const wrapper = shallow(<ListUsers />);
  //   expect(wrapper).toMatchSnapshot();
  // });

  // it("Display users", () => {
  //   jest.spyOn(React, "useState").mockImplementationOnce(() => {
  //     useState(userDetails);
  //   });
  //   jest.spyOn(React, "useEffect").mockImplementationOnce(() => {
  //     useEffect(() => {
  //       return 1;
  //     });
  //   });
  //   const wrapper = shallow(<ListUsers />);
  //   const fName = wrapper.find({ dataKey: "first-name" });
  //   console.log(fName.debug());
  //   expect(wrapper.find({ dataKey: "first-name" }).length).toBe(1);
  // });
});
