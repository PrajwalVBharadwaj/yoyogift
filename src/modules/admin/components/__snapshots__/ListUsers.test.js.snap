// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing List users component Snapshot testign 1`] = `ShallowWrapper {}`;
